### ISSUE TITLE
Navigation link: incorrect usage of Tooltip

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#68597]
### Description

Similar to https://github.com/WordPress/gutenberg/issues/66271 and https://github.com/WordPress/gutenberg/issues/68591
This is one more occurrence of incorrect usage of the Tooltip component. 

Tooltips should be used to visually expose the accessible name of an interactive control. They're not the right tool to use for descriptions or additional information.
- In the best case, the description / additional information isn't accessible to all users.
- In the worst case, the non-interactive element the Tooltip is used with will get a tabindex=0 attribute added by the Ariakit-based Tooltip component, thus making the non-interactive element focusable. This is a WCAG violation and in some cases disrupts the expected keyboard interaction.

In 

https://github.com/WordPress/gutenberg/blob/79234bd5b182a8be79129cb26db6b76a63daac0d/packages/block-library/src/navigation-link/edit.js#L536-L538

and 

https://github.com/WordPress/gutenberg/blob/79234bd5b182a8be79129cb26db6b76a63daac0d/packages/block-library/src/navigation-link/edit.js#L582-L601

The Tooltip wraps a `span` element. As such, the span element gets a tabindex=0 attribute and becomes a non-interactive focusable element.

Once again, it is worth reminding that tooltips should not be used to provide information in a 'visual' way, that is: only on mouse hover.

![Image](https://github.com/user-attachments/assets/a21fb5ee-42e5-432a-9e41-023cc0b5e4b9)
![Image](https://github.com/user-attachments/assets/af144a92-879f-4daf-958f-b66ce53860cc)
![Image](https://github.com/user-attachments/assets/8386fa91-87eb-4787-85cd-5898dafff360)

Instead of using tooltips, the invalid or missing link state should be communicated by the visible placeholder text.
For example, when a link is missing, 'Add link' should be 'Missing link' and maybe the item should be visually highlighted in some way as an error or warning.

### Step-by-step reproduction instructions

- Edit a navigation menu.
- Add a link to a post and save.
- In another browser tab, delete the linked post.
- Refresh the editor page.
- Observe a text `(Invalid)` is added after the link.
- Hover the link and observe a tooltip appears with some extra info.
- Add a link to a post or page.
- Select the newly added item in the navigation and click 'Remove link' in the popover.
- Observe the menu item text becomes 'Add link'
- Hover the link and observe a tooltip appears with some extra info.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by himanshupathak95** [comment]
Hey @afercia, Thanks for reporting this.

I resonate with the proposal and will soon be raising a PR with a fix.
